src/appleseed/renderer/modeling/postprocessingstage/renderstamppostprocessingstage.cpp

```diff
@@ -143,44 +144,48 @@ namespace
             const CanvasProperties& props = frame.image().properties();
             const float image_height = static_cast<float>(props.m_canvas_height);
             const float text_height = TextRenderer::compute_string_height(font_height, text.c_str());
-            const float origin_y = image_height - text_height - MarginV;
+            const float origin_y = image_height - text_height - margin_v;
 
             // Draw the background rectangle.
             Drawing::draw_filled_rect(
                 frame.image(),
                 Vector2i(
                     0,
-                    truncate<int>(origin_y - MarginV)),
+                    truncate<int>(origin_y - margin_v)),
                 Vector2i(
                     static_cast<int>(props.m_canvas_width - 1),
                     static_cast<int>(props.m_canvas_height - 1)),
                 BackgroundColor);
 
             // Calculate final icon size and scale the icon
             const ImageSpec icon_spec = m_icon.spec();
+            const float stamp_height = text_height + 2 * margin_v;
+            const float icon_height = 0.68f * stamp_height;
             const float aspect = static_cast<float>(icon_spec.width) / static_cast<float>(icon_spec.height);
             const ROI roi(
                 0,
-                static_cast<int>(m_scale_factor * aspect * DefaultIconHeight),
+                static_cast<int>(round(icon_height * aspect)),
```

Comment:
The `static_cast<int>()` is redundant, this should be enough:

    round<int>(icon_weight * aspect)